src/impl.cpp

```diff
@@ -0,0 +1,9 @@
+#include "impl.h"
+#include <cstring>
+#include <bitset>
+
+char* IntToBinaryRepresentation(char *buffer, int source)
+{
+    strcpy(buffer, std::bitset<(sizeof(source)*8)>(source).to_string().c_str());
```

Comment:
**Cheating :)**
Please do not use the std::bitset
(yes, this is a nice solution, but training author wants a little bit bytefucking)

Reply:
char* IntToBinaryRepresentation(char *buffer, int source)
{
//    strcpy(buffer, std::bitset<(sizeof(source)*8)>(source).to_string().c_str());
    char* buf = buffer;
    char* byte = (char *)&source;
    for( int i = sizeof(source)-1; i > -1; i--) {
        for( int j = 0; j < 8; j++) {
          if ( byte[i] & 0x80) {
              *buf = '1';
          } else {
              *buf = '0';
          }
          byte[i] <<= 1;
          buf++;
        }
    }
    *buf = '\0';
    return buffer;
}

Reply:
Please rework in your repo